pkg/asset/machines/baremetal/machines.go

```diff
@@ -60,11 +62,18 @@ func Machines(clusterID string, config *types.InstallConfig, pool *types.Machine
 	return machines, nil
 }
 
-func provider(clusterName string, networkInterfaceAddress string, platform *baremetal.Platform, userDataSecret string) *baremetalprovider.BareMetalMachineProviderSpec {
+func provider(clusterName string, networkInterfaceAddress string, platform *baremetal.Platform, osImage string, userDataSecret string) *baremetalprovider.BareMetalMachineProviderSpec {
+	// The rhcos-downloader container launched by the baremetal-operator downloads the image,
+	// compresses it to speed up deployments and makes it available on provisioningIP via http
+	provisioningIP := platform.ClusterProvisioningIP
```

Comment:
nit: this is only used once, so directly using `platform.ClusterProvisioningIP` at call site might be better.